lhotse/dataset/speech_synthesis.py

```diff
@@ -0,0 +1,73 @@
+import re
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import torch
+from torch.utils.data import Dataset
+
+from lhotse.cut import CutSet
+from lhotse.utils import Pathlike
+
+EPS = 1e-8
+
+
+def text_to_tokens(text: str) -> List[str]:
+    text = re.sub(r'[^\w !?]', '', text)
+    text = re.sub(r'^\s+', '', text)
+    text = re.sub(r'\s+$', '', text)
+    text = re.sub(' +', ' ', text)
```

Comment:
```suggestion
    text = re.sub('\s+', ' ', text)
```

Just to be on the safe side if there's any tabs or other whitespace characters. 

As a note to myself and for discussion - it seems like this is another place in the code (next to ASR dataset) where we could use some kind of standardized interface for customizable text normalization (as probably almost every corpus will require to do it a bit differently).

Reply:
Mm.  My inclination would be to leave this in the more 'local' part of the code, in ljspeech.py.
I am cautious about putting anything like that centrally until we are more confident we know where we are going with it.  The danger is that we'll be tempted to add options, etc., creating interfaces that aren't needed.

Reply:
I was thinking about how to interact with text normalization in Lhotse for some time now, and I suggest we add a method called "transform_text" to `CutSet` and `SupervisionSet` that accepts a single function argument (`transform(text: str) -> str`), goes through all supervisions and applies that on their text. This is general enough that you can use it to split to characters, sub-word tokens, transcribe with G2P, whatever you can think of.

BTW Ideally we return new objects instead of modifying the existing ones, as that would be a consistent behaviour with other transform-like methods I implemented so far. I'm not 100% sure if we want immutability for Cuts/Supervisions etc. in the long run - if things get sluggish, we should rethink.

Reply:
I think we can let users modify supervisions by their own functions, and the ljspeech recipe provide a TN function as well, as  the commit https://github.com/pzelasko/lhotse/pull/81/commits/77d07e447090ea2a9b51285b1ec9956809952046 do.